walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -62,24 +68,96 @@
     return meetingTimes;
   }
 
+  /** 
+   * Return an optimized version of possible meeting times that accommodates the maximum number of optional attendees. 
+   * @param meetingTimes available meeting slots where only mandatory attendees are considered 
+   * @param unavailableTimesOptionalAttendees unavailable times for optional attendees 
+   * @param duration the duration of the meeting to be scheduled 
+   * @return an optimized list of meeting times that would include the maximum number of attendees 
+   */
+  private List<TimeRange> getOptimiazedMeetingTimes(List<TimeRange> meetingTimes, List<TimeRange> unavailableTimesOptionalAttendees, long duration) {
+    List<TimeRange> optimizedMeetingTimes = new ArrayList<>();
+    
+    for (int i = 0; i < meetingTimes.size(); i++) {
+      TimeRange meetingSlot = meetingTimes.get(i);
+      boolean canSkipRange = false;
+      for (TimeRange t : unavailableTimesOptionalAttendees) {
```

Comment:
I didn't sort in my new implementation, but since `add` into the `MeetingRanges` "data structure" is O(n), my solution is still O(n^2). Is there a way to improve this runtime?

Reply:
I still think it could be improved, but it would likely look very different from your current implementation. This is fine to merge as-is.